mock/py/mockbuild/backend.py

```diff
@@ -567,9 +636,21 @@ def buildsrpm(self, spec, sources, timeout, follow_links):
                 else:
                     shutil.copy(sources, self.buildroot.make_chroot_path(self.buildroot.builddir, "SOURCES"))
 
-            spec = self.buildroot.make_chroot_path(self.buildroot.builddir, "SPECS", os.path.basename(spec))
+                host_chrootsources = self.buildroot.make_chroot_path(self.buildroot.builddir, "SOURCES")
+
+            host_chrootspec = self.buildroot.make_chroot_path(self.buildroot.builddir, "SPECS", os.path.basename(spec))
+
+            if self.config.get('rpkg_preprocessing'):
+                self.buildroot.root_log.info("Rpkg preprocessing enabled.")
+
+                if not host_chrootsources:
+                    self.buildroot.root_log.info(
+                        "Sources not present. Skipping rpkg preprocessing step.")
+                else:
+                    self.rpkgPreprocess(host_chrootspec, host_chrootsources, timeout)
+
             # get rid of rootdir prefix
-            chrootspec = spec.replace(self.buildroot.make_chroot_path(), '')
```

Comment:
If you did not `s/spec/host_chrootspec/`, the commit would be easier read.

Reply:
Yes, I know but I was getting confused myself with the variable names there so I thought I should make the names of variables more understandable.